crates/service/src/middleware/auth/tap.rs

```diff
@@ -65,3 +65,135 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{sync::Arc, time::Duration};
+
+    use alloy::primitives::{address, Address};
+    use axum::{
+        body::Body,
+        http::{Request, Response},
+    };
+    use prometheus::core::Collector;
+    use reqwest::StatusCode;
+    use sqlx::PgPool;
+    use tap_core::{
+        manager::Manager,
+        receipt::{
+            checks::{Check, CheckError, CheckList, CheckResult},
+            state::Checking,
+            ReceiptWithState,
+        },
+    };
+    use test_assets::{create_signed_receipt, TAP_EIP712_DOMAIN};
+    use tower::{Service, ServiceBuilder, ServiceExt};
+    use tower_http::auth::AsyncRequireAuthorizationLayer;
+
+    use crate::{
+        middleware::{
+            auth::tap_receipt_authorize,
+            metrics::{MetricLabelProvider, MetricLabels},
+        },
+        tap::IndexerTapContext,
+    };
+
+    const ALLOCATION_ID: Address = address!("deadbeefcafebabedeadbeefcafebabedeadbeef");
+
+    async fn handle(_: Request<Body>) -> anyhow::Result<Response<Body>> {
+        Ok(Response::new(Body::default()))
+    }
+
+    struct TestLabel;
+    impl MetricLabelProvider for TestLabel {
+        fn get_labels(&self) -> Vec<&str> {
+            vec!["label1"]
+        }
+    }
+
+    #[sqlx::test(migrations = "../../migrations")]
+    async fn test_tap_middleware(pgpool: PgPool) {
```

Comment:
This is a great test, but could you split this in three? `test_valid_receipt`, `test_invalid_receipt_with_failed_metric`, and `test_missing_signed_receipt`